### PR TITLE
RavenDB-17381 Transaction recorder error handling

### PR DESCRIPTION
**Pull request by Danielle9897**
### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-17381

### Additional description
Studio: fixed placeholder + added available directories
Server: added validation

### Type of change
- Bug fix

### How risky is the change?
- Low 

### Backward compatibility
- Non breaking change

### Is it platform specific issue?
- No

### Documentation update
- No documentation update is needed 

### Testing 
- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?
- No

### UI work
- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/TransactionsRecordingHandler.cs [hunk 1]
```diff
@@ -146,11 +146,24 @@ public async Task StartRecording()
...
 
+                if (File.Exists(outputFilePath))
+                {
+                    throw new InvalidOperationException("File " + outputFilePath + " already exists");
+                }
+
+                // here path is either a new file -or- an existing directory
+                if (Directory.Exists(outputFilePath))
+                { 
+                    throw new InvalidOperationException("Please enter a path including the file name");
```
Comment (ayende) [inline src/Raven.Server/Documents/Handlers/TransactionsRecordingHandler.cs]:
```suggestion
                    throw new InvalidOperationException(outputFilePath + " is a directory, please enter a path to a file.");
```

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Handlers/TransactionsRecordingHandler.cs (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Server/Documents/Handlers/TransactionsRecordingHandler.cs]: done

File: src/Raven.Studio/typescript/viewmodels/manage/debugAdvancedRecordTransactionCommands.ts [hunk 2]
```diff
@@ -30,22 +31,36 @@ class debugAdvancedRecordTransactionCommands extends viewModelBase {
...
+    outputFilePathHasFocus = ko.observable<boolean>(false);
     
     runningTasks = ko.observableArray<runningRecording>([]);
     
     validationGroup: KnockoutValidationGroup;
     
     // marker for parent router
     static preventParentGrow = true;
+
+    outputFilePathOptions = ko.observableArray<string>(["aaa"]);
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/manage/debugAdvancedRecordTransactionCommands.ts]: aaa?

File: src/Raven.Studio/typescript/viewmodels/manage/debugAdvancedRecordTransactionCommands.ts [hunk 3]
```diff
@@ -30,22 +31,36 @@ class debugAdvancedRecordTransactionCommands extends viewModelBase {
...
+        _.bindAll(this, "outputFilePathChanged");
+        this.updateOutputFilePathOptions(this.outputFile());
         
+        this.initObservables();
         this.initValidation();
     }
     
-    private initValidation() {
+    private initObservables(): void {
+        this.outputFile.throttle(300).subscribe((newPathValue) => {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/manage/debugAdvancedRecordTransactionCommands.ts]: no need for `(` `)`

Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/manage/debugAdvancedRecordTransactionCommands.ts]: done

File: src/Raven.Studio/typescript/viewmodels/manage/debugAdvancedRecordTransactionCommands.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/manage/debugAdvancedRecordTransactionCommands.ts]: oops - forgot to remove...

**Review by ml054 — APPROVED** [review]

**Pull request merged**